### DISCUSSION TITLE
Allow for ABIs to be generated from external contracts

### DISCUSSION THREAD

**Discussion by coreyar** [wevm/wagmi#3486] [Ideas]
Projects will frequently use contracts from third parties installed via a node package manager. Tools like `hardhat-deploy` support this natively allowing you to reference contracts and artifacts not defined in the current hardhat project.

It would be useful if the hardhat plugin was extended to allow for an array of artifacts directories that can be included in abi generation. This would allow for supporting third part contracts.

**Comment by tmm** [comment]
Tighter integration would be nice, but no plans at the moment for the core team to work on that (have some higher priority items on the roadmap).

In the meantime, check out [Async CLI Config](https://wagmi.sh/cli/config/configuring-cli#async-config). You can read artifacts from the file system and pass them to the [`contracts`](https://wagmi.sh/cli/config/options#contracts) option directly.

**Comment by coreyar** [comment]
Cool. I'll take a look. Thanks!
